lib/iaas/AzureClient.js

```diff
@@ -0,0 +1,192 @@
+      .catch(BaseCloudClient.providerErrorTypes.Unauthorized, err => {
+        logger.error(err.message);
+        throw new Unauthorized('Authorization at the cloud storage provider failed');
+      })
+      .catch(BaseCloudClient.providerErrorTypes.Forbidden, err => {
+        logger.error(err.message);
+        throw new Forbidden('Authentication at the cloud storage provider failed');
+      })
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${file}' not found`));
+  }
+
+  download(options) {
+    return utils.streamToPromise(this.storage.createReadStream(
+        options.container, options.remote))
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${options.remote}' not found`));
+  }
+
+  upload(options, buffer) {
+    return new Promise((resolve, reject) => {
+      function cleanup() {
+        uploadedStream.removeListener('finish', onfinish);
+        uploadedStream.removeListener('error', onerror);
+      }
+
+      function onerror(err) {
+        cleanup();
+        reject(err);
+      }
+
+      function onfinish() {
+        cleanup();
+        resolve(JSON.parse(buffer));
+      }
+
+      let uploadedStream = this.storage
+        .createWriteStreamToBlockBlob(
+          options.container, options.remote);
+
+      uploadedStream.on('error', onerror);
+      uploadedStream.on('finish', onfinish);
+      uploadedStream.end(buffer);
+    });
+  }
+
+  uploadJson(container, file, data) {
+    if (arguments.length < 3) {
+      data = file;
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .upload({
+        container: this.containerName,
+        remote: file
+      }, new Buffer(JSON.stringify(data, null, 2), 'utf8'));
+  }
+
+  downloadJson(container, file) {
+    if (arguments.length < 2) {
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .download({
+        container: this.containerName,
+        remote: file
+      })
+      .then((data) => {
+        const response = JSON.parse(data);
+        response.trigger = response.trigger === CONST.BACKUP.TRIGGER.MANUAL ? CONST.BACKUP.TRIGGER.SCHEDULED : response.trigger;
+        //The above conversion is done to handle existing CRON Jobs which set this trigger as 'manual' even for scheduled Jobs
+        //Above conversion can be removed and code changes can be revereted 14 days after the current fix goes live
+        return response;
+      })
+      .catchThrow(SyntaxError, new NotFound(`Object '${file}' not found`));
+  }
+
+  deleteSnapshot(snapshotName) {
+    return Promise
+      .try(() => {
+        return this.computeClient
+          .snapshots
+          .deleteMethodAsync(this.settings.resource_group, snapshotName)
+          .then(retval => logger.info(`Deleted snapshot ${snapshotName}`, retval))
```

Comment:
what is retVal ? is it a boolean? In that case log this in a tap as the response from deleteSnapshot will not be a boolean but undefined because of above logger. In either case better to use tap if you just want to log response.

Reply:
As discussed .... Don't want to return response to caller. Will keep as it is.

Reply:
ok